### ISSUE TITLE
Inheriting NeoGraphItemsProducers

### ISSUE THREAD

**Issue by larsw** [dunnock/react-sigma#48]
Hi,

Should it be possible to inherit or overload the two factory functions (node() & edge() ) in this class?
I ended up monkey patching it by adding decorators on a runtime instance by adjusting the prototype - pretty nasty stuff.

```javascript
    this.myProducers = new NeoGraphItemsProducers()
    const myProto = Object.getPrototypeOf(this.myProducers)

    const baseNodeFactory = myProto.node.bind(this.myProducers)
    const baseEdgeFactory = myProto.edge.bind(this.myProducers)

    myProto.node = (n) => {
      var x = baseNodeFactory(n)
      x.label = n.properties.Name
      return x
    }

    myProto.edge = (e) => {
      var x = baseEdgeFactory(e)
      x.label = e.properties.Name
      return x
    }

    Object.setPrototypeOf(this.myProducers, myProto)

```
I tried to extend the ES6 version of the `NeoGraphItemsProducers` class, but when I try to instantiate the inherited class, I end up with `TypeError: Class constructor NeoGraphItemsProducers cannot be invoked without 'new'`

**Comment by dunnock** [comment]
Hi @larsw,

NeoGraphItemsProducers has nothing valuable inside which can be beneficial for inheritnce but interface, since JS does not have interfaces anyways that would be better just to create new class:

```
class MyGraphItemsProducers {
  constructor() {}
  node = (n) => {
      var x = baseNodeFactory(n)
      x.label = n.properties.Name
      return x
    }
    edge = (e) => {
      var x = baseEdgeFactory(e)
      x.label = e.properties.Name
      return x
    }
}
```

Then you can pass it to `<NeoCypher producers={new MyGraphItemsProducers()} .../>`

Hope it helps

**Comment by larsw** [comment]
Tried that; problem is that the current PropType checking nags about the type of the input type if you create your own (maybe you can lax the PropType check for the `producers` prop?

**Comment by dunnock** [comment]
Fixed in react-sigma@1.2.28

**Comment by larsw** [comment]
Thanks!
